app/services/oregon_digital/collection_streamer.rb

```diff
@@ -67,7 +45,7 @@ def stream_child_collections(collection, zip, folder, keys, controlled_keys)
     def stream_child_works(collection, zip, folder, keys, controlled_keys)
       collection.child_works.map do |work|
         # Add low quality works from collection and append metadata
-        stream_works_low(work, zip, folder)
+        stream_works_low(work, zip, folder) if work.visibility == 'open'
```

Comment:
I want to test this line. I see it's what was originally used to check if the work should be downloadable, but I don't think this is sufficient anymore. I'm thinking this might leak OSU/UO only works to non-OSU or non-UO users.
This might need to change to something like `user.can?(:download, work)`.
Let me run a test and report back here

Reply:
So yes, It looks like we're going to want to make sure the work is actually `:download_low` allowed for the user.
That'll look something like `next unless user.can?(:download_low, work)`

You'll probably have to do some finagling with `OregonDigital::Ability::WorkDownloadAbility` and the structure of `OregonDigital::CollectionStreamer` to include a user object